src/state/slices/portfolioSlice/portfolioSlice.test.ts

```diff
@@ -19,456 +38,746 @@ import {
+            tokens: [mockEthToken({ balance: '200100000000000000' })]
+          }
         }
-      }
+      })
+
+      // dispatch portfolio data
+      store.dispatch(
+        portfolioSlice.actions.upsertPortfolio(mockUpsertPortfolio([ethAccount, ethAccount2]))
+      )
+
+      const ethAccount2Id = `${ethAccount2.caip2}:${toLower(ethAccount2.pubkey)}`
+
+      // dispatch market data
+      const ethMarketData = mockMarketData({
+        price: '1000'
+      })
+
+      const foxMarketData = mockMarketData({
+        price: '10'
+      })
+
+      const usdcMarketData = mockMarketData({
+        price: '1'
+      })
+
+      store.dispatch(
+        marketDataSlice.actions.setMarketData({
+          [ethCaip19]: ethMarketData,
+          [foxCaip19]: foxMarketData,
+          [usdcCaip19]: usdcMarketData
+        })
+      )
 
-      const returnValue = {
-        [ethAccountSpecifier1]: {
-          [ethCaip19]: '0.00',
-          [foxCaip19]: '0.00'
+      // dispatch asset data
+      const assetData = mockAssetState()
+      store.dispatch(assetsSlice.actions.setAssets(assetData))
+      const state = store.getState()
+
+      it('should be able to filter by assetId', () => {
+        const expected = '1.200009'
+        const result = selectPortfolioCryptoHumanBalanceByFilter(state, { assetId: ethCaip19 })
+        expect(result).toEqual(expected)
+      })
+
+      it('should be able to filter by accountId and assetId', () => {
+        const expected = '0.2001'
+        const result = selectPortfolioCryptoHumanBalanceByFilter(state, {
+          accountId: ethAccount2Id,
+          assetId: foxCaip19
+        })
+        expect(result).toEqual(expected)
+      })
+    })
+
+    describe('selectPortfolioTotalFiatBalanceByAccount', () => {
+      const store = createStore()
+      const { ethAccount, ethAccount2 } = mockETHandBTCAccounts({
+        ethAccountObj: {
+          balance: '1000000000000000000',
+          chainSpecific: {
+            tokens: [
+              mockEthToken({ balance: '1000000000000000000', caip19: foxCaip19 }),
+              mockEthToken({ balance: '1000000', caip19: usdcCaip19 })
+            ]
+          }
         },
-        [ethAccountSpecifier2]: {
-          [ethCaip19]: '0.00',
-          [foxCaip19]: '0.00'
+        ethAccount2Obj: {
+          balance: '2000000000000000000'
         }
-      }
+      })
+
+      // dispatch portfolio data
+      store.dispatch(
+        portfolioSlice.actions.upsertPortfolio(mockUpsertPortfolio([ethAccount, ethAccount2]))
+      )
+
+      const ethAccountId = `${ethAccount.caip2}:${toLower(ethAccount.pubkey)}`
+      const ethAccount2Id = `${ethAccount2.caip2}:${toLower(ethAccount2.pubkey)}`
+
+      // dispatch market data
+      const ethMarketData = mockMarketData({
+        price: '1000'
+      })
+
+      const foxMarketData = mockMarketData({
+        price: '10'
+      })
+
+      const usdcMarketData = mockMarketData({
+        price: '1'
+      })
```

Comment:
You use this same market data in a bunch of places so why no turn 11 lines into 3?
```suggestion
      const ethMarketData = mockMarketData({ price: '1000' })
      const foxMarketData = mockMarketData({ price: '10' })
      const usdcMarketData = mockMarketData({ price: '1' })
```

Reply:
Good call